### ISSUE TITLE
Error: Invalid ref: #/components/schemas/...

### ISSUE THREAD

**Issue by andreas-soroko** [astahmer/openapi-zod-client#44]
Hi,

just tried to used it on a existing service and got the error
```
Error: Invalid ref: #/components/schemas/NameResolutionType2

{
  ref: '#/components/schemas/NameResolutionType2',
  fallbackName: 'nameResolutionType',
  result: 'NameResolutionType2.optional().default("AssetName")'
}

```

but don't get it, what is wrong. 

minimal reproduction example:
```json
{
  "x-generator": "NSwag v13.17.0.0 (NJsonSchema v10.8.0.0 (Newtonsoft.Json v13.0.0.0))",
  "openapi": "3.0.0",
  "info": {
    "title": "title",
    "version": "1.5.0.0"
  },
  "paths": {
    "/api/v2/name-resolution/via-id/{assetId}": {
      "get": {
        "tags": [
          "NameResolutionApi"
        ],
        "operationId": "api_v2_name-resolution_via-id_{assetid}_get_ResolveNamesForAssetV2",
        "parameters": [
          {
            "name": "nameResolutionType",
            "in": "query",
            "schema": {
              "default": "AssetName",
              "oneOf": [
                {
                  "$ref": "#/components/schemas/NameResolutionType2"
                }
              ]
            }
          }
        ],
        "responses": {
          "200": {
            "description": "",
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "type": "string"
                  }
                }
              }
            }
          },
          "400": {
            "description": ""
          },
          "401": {
            "description": "Unauthorized"
          },
          "403": {
            "description": "Forbidden"
          },
          "404": {
            "description": ""
          },
          "500": {
            "description": ""
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "NameResolutionType2": {
        "type": "string",
        "description": "",
        "x-enumNames": [
          "AssetName",
          "FullName"
        ],
        "enum": [
          "AssetName",
          "FullName"
        ]
      }
    }
  }
}
```

**Comment by astahmer** [comment]
Hey, my bad it's a regression caused by the chains (`.optional().default("AssetName")`) following the ref (`NameResolutionType2`) that I haven't tested, I just published a new version with a fix (v1.4.1)

related commit:
https://github.com/astahmer/openapi-zod-client/commit/17a7091842220091a11d3586cec92299a5d5fbcd

**Comment by andreas-soroko** [comment]
wow, that was a fast fix - thank you! its working
